lms/djangoapps/mobile_api/testutils.py

```diff
@@ -123,8 +137,150 @@ def test_other_user(self):
         self.api_response(expected_response_code=404, username=other.username)
 
 
+class MobileAPIMilestonesMixin(object):
+    """
+    Tests the Mobile API decorators for milestones.
+
+    The two milestones supported in these tests are entrance exams and
+    pre-requisite courses. If either of these milestones are unfulfilled,
+    the mobile api will appropriately block content until the milestone is
+    fulfilled.
+    """
+    MILESTONE_MESSAGE = {
+        'developer_message':
+            'Cannot access content with unfulfilled pre-requisites or unpassed entrance exam.'
+    }
+
+    ALLOW_ACCESS_TO_MILESTONE_COURSE = False  # pylint: disable=invalid-name
+
+    def _add_entrance_exam(self):
+        """ Sets up entrance exam """
+        # Set up the extrance exam
+        self.course.entrance_exam_enabled = True
+
+        self.entrance_exam = ItemFactory.create(  # pylint: disable=attribute-defined-outside-init
+            parent=self.course,
+            category="chapter",
+            display_name="Entrance Exam Chapter",
+            is_entrance_exam=True,
+            in_entrance_exam=True
+        )
+        self.problem_1 = ItemFactory.create(  # pylint: disable=attribute-defined-outside-init
+            parent=self.entrance_exam,
+            category='problem',
+            display_name="The Only Exam Problem",
+            graded=True,
+            in_entrance_exam=True
+        )
+
+        add_entrance_exam_milestone(self.course, self.entrance_exam)
+
+        self.course.entrance_exam_minimum_score_pct = 0.50
+        self.course.entrance_exam_id = unicode(self.entrance_exam.location)
+        modulestore().update_item(self.course, self.user.id)
+
+    def _add_prerequisite_course(self):
+        """ Helper method to set up the prerequisite course """
+        self.prereq_course = CourseFactory.create()  # pylint: disable=attribute-defined-outside-init
+        add_prerequisite_course(self.course.id, self.prereq_course.id)
+
+    def _pass_entrance_exam(self):
+        """ Helper function to pass the entrance exam """
+        # set up the request for exam functions
+        request = create_mock_request(self.user)
+        answer_entrance_exam_problem(self.course, request, self.problem_1)
+
+    def verify_response(self):
```

Comment:
Please change the name of this method to something like: `verify_unfulfilled_milestone_response`.  Otherwise, from looking at the code that calls it, it is confusing on how it differs from `api_response`.